core/api/views/business_plan.py

```diff
@@ -101,13 +103,30 @@ def create(self, request, *args, **kwargs):
         self.perform_create(serializer)
         instance = serializer.instance
 
+        # check user permissions
+        user = request.user
+        if (
+            user.agency_role == user.AgencyRole.AGENCY_INPUTTER
+            and instance.status != BusinessPlan.Status.draft
+        ):
```

Comment:
In the future we should make sure that a user can create only draft versions of bp. We should not let the user create from the beginning final versions. Maybe add a comment about this here just to make sure we do not forget to make this change